### ISSUE TITLE
How to we update the learning on every step?

### ISSUE THREAD

**Issue by louis2889184** [PyTorchLightning/pytorch-lightning#5584]
#### What is your question?

I want to update the learning on training steps. However, I found that 
https://github.com/PyTorchLightning/pytorch-lightning/blob/f477c2fd2980ad128bfe79a3b859e0b81b435507/pytorch_lightning/trainer/optimizers.py#L97-L108
the interval is set as `epoch`. It means the update will always come after each `epoch`, even though there is a line to update the learning rate on steps at https://github.com/PyTorchLightning/pytorch-lightning/blob/f477c2fd2980ad128bfe79a3b859e0b81b435507/pytorch_lightning/trainer/training_loop.py#L573
This line will never actually happen because the scheduler's `interval` (`epoch`) is different from the argument `interval` (`step`).

Maybe I misunderstood. If yes, how could I update the learning rate on every step?

**Comment by carmocca** [comment]
> However, I found that [...]

What you are referencing there is an implementation detail to create the default scheduler configuration. That configuration is not forced

You can define `configure_optimizers` with any of the options described [here](https://pytorch-lightning.readthedocs.io/en/stable/optimizers.html#learning-rate-scheduling)

In your case, setting the scheduler key `'interval': 'step'`

**Comment by louis2889184** [comment]
That's it! Thank you.
